src/python/pants/engine/internals/graph.py

```diff
@@ -168,85 +175,170 @@ def warn_deprecated_field_type(request: _WarnDeprecatedFieldRequest) -> _WarnDep
     return _WarnDeprecatedField()
 
 
+@dataclass(frozen=True)
+class _TargetParametrizations:
+    """All parametrizations and generated targets for a single input Address.
+
+    If a Target has been parametrized, it might _not_ be present in this output, due to it not being
+    addressable using its un-parameterized Address.
+    """
+
+    target: Target | None
```

Comment:
What does this represent? The original target, right? A more verbose name like `original_tgt` would be helpful.

Reply:
Sounds good. I'll also make it non-optional in this change, and move the optionality to #14408.